lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/api/DataLoader.java

```diff
@@ -100,7 +100,7 @@ public DataLoader(
       @Parameter(DataLoadingRequestBuilder.DataLoadingComputeRequest.class) final String serializedComputeRequest) {
     this(clock, requestor, dataLoadingService, new HashSet<String>(
         Arrays.asList(serializedComputeRequest)), new HashSet<String>(
-        Arrays.asList(EvaluatorRequestSerializer.serialize(EvaluatorRequest
+        Arrays.asList(AvroEvaluatorRequestSerializer.serialize(EvaluatorRequest
```

Comment:
If you change the AvroEvaluatorRequestSerializer as injectable, here you can simply put AvroEvaluatorRequestSerializer as a parameter in the constructor and use it inside the constructor. You don't need to worry about who will create the instance of it.